skglm/solvers/fista.py

```diff
@@ -0,0 +1,93 @@
+import numpy as np
+from scipy.sparse import issparse
+from numba import njit
+from skglm.solvers.base import BaseSolver
+from skglm.solvers.common import construct_grad, construct_grad_sparse
+
+
+@njit
+def _prox_vec(w, z, penalty, lipschitz):
+    n_features = w.shape[0]
+    for j in range(n_features):
+        w[j] = penalty.prox_1d(z[j], 1 / lipschitz, j)
+    return w
+
+
+class FISTA(BaseSolver):
+    r"""ISTA solver with Nesterov acceleration (FISTA).
+
+    This solver implements accelerated proximal gradient descent for linear problems.
+
+    Attributes
+    ----------
+    max_iter : int, default 100
+        Maximum number of iterations.
+
+    tol : float, default 1e-4
+        Tolerance for convergence.
+
+    opt_freq : int, default 10
```

Comment:
Is it useful to expose the param to control the frequency of optimality check?

Reply:
I removed it.